File: src/dataset/go/sublist_Exercism_go_drapala/openAI/ChatGPT4_sublist_v4.go
package sublist

type Relation = string

func isSublist(sm, bg []int) bool {
	smLen, bgLen := len(sm), len(bg)
	for i := 0; i <= bgLen-smLen; i++ {
		match := true
		for j := 0; j < smLen; j++ {
			if sm[j] != bg[i+j] {
				match = false
				break
			}
		}
		if match {
			return true
		}
	}
	return false
}

func Sublist(l1, l2 []int) string {
	len1, len2 := len(l1), len(l2)

	switch {
	case len1 == 0 && len2 == 0:
		return "equal"
	case len1 == 0:
		return "sublist"
	case len2 == 0:
		return "superlist"
	case len1 == len2 && isSublist(l1, l2):
		return "equal"
	case len1 < len2 && isSublist(l1, l2):
		return "sublist"
	case len1 > len2 && isSublist(l2, l1):
		return "superlist"
	default:
		return "unequal"
	}
}

File: src/dataset/typescript/sublist_Exercism-typescript-FilipeCerejo/openAI/ChatGPT4_sublist_v4.ts
export class List {
    list: number[];

    constructor(...list: number[]) {
        this.list = list;
    }

    compare(otherList: List): 'superlist' | 'sublist' | 'equal' | 'unequal' {
        const len1 = this.list.length;
        const len2 = otherList.list.length;

        if (len1 === len2) {
            return this.areEqual(this.list, otherList.list) ? 'equal' : 'unequal';
        }

        if (len1 > len2) {
            return this.isSublist(this.list, otherList.list) ? 'superlist' : 'unequal';
        }

        return this.isSublist(otherList.list, this.list) ? 'sublist' : 'unequal';
    }

    private isSublist(bigger: number[], smaller: number[]): boolean {
        const smallerLen = smaller.length;
        if (smallerLen === 0) return true;

        for (let i = 0; i <= bigger.length - smallerLen; i++) {
            if (this.areEqual(bigger.slice(i, i + smallerLen), smaller)) {
                return true;
            }
        }

        return false;
    }

    private areEqual(arr1: number[], arr2: number[]): boolean {
        if (arr1.length !== arr2.length) return false;
        for (let i = 0; i < arr1.length; i++) {
            if (arr1[i] !== arr2[i]) return false;
        }
        return true;
    }
}